File: FlowCrypt/src/main/java/com/flowcrypt/email/util/exception/ApiException.java
/*
 * © 2016-2018 FlowCrypt Limited. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.util.exception;

import com.flowcrypt.email.api.retrofit.response.base.ApiError;

import java.io.IOException;

/**
 * An API exception.
 *
 * @author Denis Bondarenko
 *         Date: 30.01.2018
 *         Time: 18:39
 *         E-mail: DenBond7@gmail.com
 */

public class ApiException extends IOException {
    public ApiException(ApiError apiError) {
        super("API error: code = " + apiError.getCode() + ", message = " + apiError.getMessage());
    }
}

File: FlowCrypt/src/main/java/com/flowcrypt/email/util/exception/ApiException.kt
/*
 * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.util.exception

import com.flowcrypt.email.api.retrofit.response.base.ApiError

import java.io.IOException

/**
 * An API exception.
 *
 * @author Denys Bondarenko
 */
class ApiException(val apiError: ApiError?) :
  IOException("API error: code = " + apiError?.code + ", message = " + apiError?.msg)
